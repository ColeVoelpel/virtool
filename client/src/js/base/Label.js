import styled from "styled-components";
import { getColor } from "../app/theme";

export const getLabelColor = props => getColor(props) || props.theme.color.greyDark;

export const getContrastColor = props => {
    const red = parseInt(props.color.substr(1, 2), 16);
    const green = parseInt(props.color.substr(3, 2), 16);
    const blue = parseInt(props.color.substr(5, 2), 16);
    const yiq = (red * 299 + green * 587 + blue * 114) / 1000;
    return yiq >= 128 ? "black" : "white";
};

export const Label = styled.span`
    background-color: ${getLabelColor};
    border-radius: ${props => props.theme.borderRadius.sm};
    color: ${getContrastColor};
    display: inline;
    font-size: ${props => props.theme.fontSize.sm};
    font-weight: bold;
    padding: 3px 5px;
    text-align: center;
    white-space: nowrap;
    vertical-align: baseline;

    ${props => (props.spaced ? "margin-right: 5px;" : "")}

    &:last-of-type {
        margin: 0;
    }
`;
