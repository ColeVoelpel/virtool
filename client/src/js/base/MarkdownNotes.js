import React from "react";
import styled from "styled-components";
import Marked from "marked";
import { Box } from "./Box";

const NotesBox = styled(Box)`
    overflow-y: scroll;
    max-height: 375px;
    margin-bottom: 0px;
`;

export const MarkdownNotes = ({ notes }) => {
    let markedNotes = Marked(notes ? notes : "Edit the sample to add additional notes.");
    return (
        <NotesBox>
            <div dangerouslySetInnerHTML={{ __html: markedNotes }} />
        </NotesBox>
    );
};
