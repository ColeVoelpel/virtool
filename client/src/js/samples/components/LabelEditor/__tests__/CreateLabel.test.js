import { CreateLabel } from "../CreateLabel";
import { Modal } from "../../../../base";

describe("<CreateLabel>", () => {
    let props;
    beforeEach(() => {
        props = {
            labelName: "Foobar",
            color: "#FFF000",
            description: "FooDesc",
            show: false,
            onHide: jest.fn()
        };
    });

    it("should render when [show=false]", () => {
        const wrapper = shallow(<CreateLabel {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [show=true]", () => {
        props.show = true;
        const wrapper = shallow(<CreateLabel {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should call onHide() when CreateLabel.onHide() is called", () => {
        const wrapper = shallow(<CreateLabel {...props} />);
        wrapper.find(Modal).props().onHide();
        expect(props.onHide).toHaveBeenCalled();
    });
});
