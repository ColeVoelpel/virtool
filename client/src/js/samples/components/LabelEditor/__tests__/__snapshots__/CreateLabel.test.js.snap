// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateLabel> should render when [show=false] 1`] = `
<Modal
  label="Create Label"
  onHide={[MockFunction]}
  show={false}
>
  <ModalHeader>
    Create a label
  </ModalHeader>
  <Modal__ModalBody>
    <form>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          error=""
          name="labelName"
          onChange={[Function]}
          placeholder="Label name"
          value="Foobar"
        />
        <Input__InputError />
        <Input__InputLabel>
          Description
        </Input__InputLabel>
        <Input
          name="description"
          onChange={[Function]}
          placeholder="Description"
          value="FooDesc"
        />
      </InputGroup>
      <ColorSelector
        color="#FFF000"
        errorColor=""
        name="color"
        onColorChange={[Function]}
      />
      <Button
        color="green"
        disabled={false}
        onClick={[Function]}
        tipPlacement="top"
        type="button"
      >
        Create
      </Button>
    </form>
  </Modal__ModalBody>
</Modal>
`;

exports[`<CreateLabel> should render when [show=true] 1`] = `
<Modal
  label="Create Label"
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader>
    Create a label
  </ModalHeader>
  <Modal__ModalBody>
    <form>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          error=""
          name="labelName"
          onChange={[Function]}
          placeholder="Label name"
          value="Foobar"
        />
        <Input__InputError />
        <Input__InputLabel>
          Description
        </Input__InputLabel>
        <Input
          name="description"
          onChange={[Function]}
          placeholder="Description"
          value="FooDesc"
        />
      </InputGroup>
      <ColorSelector
        color="#FFF000"
        errorColor=""
        name="color"
        onColorChange={[Function]}
      />
      <Button
        color="green"
        disabled={false}
        onClick={[Function]}
        tipPlacement="top"
        type="button"
      >
        Create
      </Button>
    </form>
  </Modal__ModalBody>
</Modal>
`;
